### PR TITLE
feat($build): Added the support for PWA

### PR DESCRIPTION
**Pull request by halfzebra**
This PR enables us to deploy apps as Progressive Web Apps.

Added SWPrecacheWebpackPlugin and a service worker to serve assets from the local cache.

@fobos @gyzerok @jackfranklin got an opinion?

I just have been to the conference, and I think this is what we should support.

Here's an example of working PWA in Elm that I've made yesterday https://halfzebra.github.io/elm-scrum-cards-pwa/

### CODE REVIEW

**Comment by gyzerok** [comment]
Seems like a good addition.

One thing that would be nice to have - docs about how to opt-out - what files to remove, what code to delete. I believe they have it described in Create React App.

It might be even better to have some automatic way to opt-out. The benefit would be that you can't mess up by forgetting to do some step. Not sure though how it would look like and how feasible it is. Just an idea.

**Comment by halfzebra** [comment]
@gyzerok thanks, I will add the docs and probably merge this 🙂

**Comment by jackfranklin** [comment]
Nice! I agree that having an opt out would be useful (for some quick apps I build I wouldn't care enough to want this) but love that we have this in!

**Comment by fobos** [comment]
Just have read the docs. Looks good.
👍

File: README.md [hunk 1]
```diff
@@ -48,20 +48,23 @@ Create a new `my-app` folder with files for your future project.
...
+│   ├── logo.svg
+│   └── manifest.json
+├── src
+│   ├── Main.elm
+│   ├── index.js
+│   ├── main.css
+│   └── registerServiceWorker.js
+└── tests
+    ├── Tests.elm
+    └── elm-package.json
```
Comment (fobos) [inline README.md]: Cool! 👍

Comment (halfzebra) [reply README.md]: @fobos thanks for checking that out!

**Pull request merged**